transaction_controller.go

```diff
@@ -0,0 +1,48 @@
+package amqp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-amqp/internal/encoding"
+)
+
+type TransactionControllerOptions struct {
+	// Capabilities is the list of extension capabilities the sender supports.
+	Capabilities []string
+}
+
+// TransactionController can interact with the transaction coordinator
+// for a transactional resource.
+// Reference: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#section-coordination
+type TransactionController struct {
+	sender *Sender
+}
+
+func (tc *TransactionController) Discharge(ctx context.Context, discharge TransactionDischarge, opts *SendOptions) error {
+	return tc.sender.Send(ctx, &Message{
+		Value: discharge,
+	}, opts)
+}
+
+func (tc *TransactionController) Declare(ctx context.Context, declare TransactionDeclare, opts *SendOptions) (any, error) {
```

Comment:
This should take a `DeclareOptions` type so it can independently evolve.

If you're confident that `Discharge` and `Declare` can take a unified options type, then maybe they can both take a `TransactionOptions` type. I'm hesitant to go down this path though.

Reply:
Separate options types is fine (with SenderOptions embedded, would be my guess)

Reply:
At present, `SendOptions` allows for sending messages as settled.  Does that apply in this context?  I haven't looked at the spec.

Reply:
That's a great point. I actually think I should _omit_ the sender options here instead. It'll just be an empty options struct.